### ISSUE TITLE
build error

### ISSUE THREAD

**Issue by VictorEijkhout** [dealii/dealii#15946]
```
[ 66%] Built target object_grid_release
[ 66%] Built target object_base_release
[ 66%] Built target object_lac_release
make[2]: *** [source/matrix_free/CMakeFiles/object_matrix_free_release.dir/all] Error 2
make[1]: *** [all] Error 2
```
[dealii_logs.zip](https://github.com/dealii/dealii/files/12479924/dealii_logs.zip)

**Comment by VictorEijkhout** [comment]
I have set the environment variable `PKG_CONFIG_PATH` to include:
```
[c206-018 dealii:5] ls /opt/intel/oneapi/tbb/2021.9.0/lib/pkgconfig
tbb.pc
```
so you should be able to find it yourself. 

Is this the right syntax?
```
cmake    -D DEAL_II_TBB_INCLUDE_DIR=/opt/intel//oneapi/tbb/latest/lib/pkgconfig/../../include/ \
```
?

**Comment by VictorEijkhout** [comment]
Nope, just `-D TBB...`

**Comment by VictorEijkhout** [comment]
Successful build. Yay. Please check for existince of `tbb.pc`?

**Comment by LemonBoy68** [comment]
> I have set the environment variable `PKG_CONFIG_PATH` to include:
> 
> ```
> [c206-018 dealii:5] ls /opt/intel/oneapi/tbb/2021.9.0/lib/pkgconfig
> tbb.pc
> ```
> 
> so you should be able to find it yourself.
> 
> Using `-D TBB_INCLUDE_DIR=.....`


-------------------------------------------------
ls /opt/intel/oneapi/tbb/2021.9.0/lib/pkgconfig
tbb32.pc  tbb.pc
-------------------------------------------------

**Comment by LemonBoy68** [comment]
I'm surprised why building with candi gives the following error, which shouldn't be the case since mpicc is already found

-- Include /work/nnh/Aspect/dealii-candi/tmp/unpack/deal.II-v9.5.1/cmake/configure/configure_10_mpi.cmake
-- Found MPI_C: /share/intel/oneapi/mpi/2021.7.0/lib/release/libmpi.so (found version "3.1")
-- Could NOT find MPI_CXX (missing: MPI_CXX_LIB_NAMES MPI_CXX_HEADER_DIR MPI_CXX_WORKS)
-- Found MPI_Fortran: /share/intel/oneapi/mpi/2021.7.0/bin/mpif90 (found version "3.1")
-- Could NOT find MPI (missing: MPI_CXX_FOUND) (found version "3.1")
-- Found MPI_MPI_H
-- Performing Test MPI_UNDERLINKAGE_OK
-- Performing Test MPI_UNDERLINKAGE_OK - Success
-- Processing MPI variables and targets
-- Unable to process MPI
-- DEAL_II_WITH_MPI has unmet external dependencies.
CMake Error at cmake/configure/configure_10_mpi.cmake:68 (message):

**Comment by masterleinad** [comment]
For forcing to use an external `TBB`, you should try setting `DEAL_II_ALLOW_BUNDLED=OFF` and `DEAL_II_FORCE_BUNDLED_*=ON` for all other bundled packages you need. If you request `DEAL_II_WITH_TBB=ON` and the external dependency is not found, you will get a configure-time error. You will likely have to set
```
-DTBB_DIR=/opt/intel/oneapi/tbb/2021.9.0
```

**Comment by LemonBoy68** [comment]
> For forcing to use an external `TBB`, you should try setting `DEAL_II_ALLOW_BUNDLED=OFF` and `DEAL_II_FORCE_BUNDLED_*=ON` for all other bundled packages you need. If you request `DEAL_II_WITH_TBB=ON` and the external dependency is not found, you will get a configure-time error. You will likely have to set
> 
> ```
> -DTBB_DIR=/opt/intel/oneapi/tbb/2021.9.0
> ```

Thanks for the answer. Is this the case?
------------------------
-D DEAL_II_ALLOW_BUNDLED=OFF \
-D DEAL_II_FORCE_BUNDLED_*=ON \
-D DEAL_II_WITH_TBB=ON \
-D TBB_DIR=/opt/intel/oneapi/tbb/2021.9.0 \
-D TBB_LIBRARY=/opt/intel/oneapi/tbb/latest/lib/intel64/gcc4.8/libtbb.so \
-D TBB_DEBUG_LIBRARY=/opt/intel/oneapi/tbb/latest/lib/intel64/gcc4.8/libtbb_debug.so \
------------------------

**Comment by masterleinad** [comment]
No, the "*" in `-D DEAL_II_FORCE_BUNDLED_*=ON` was a placeholder for any bundled packages you still need (like `-DDEAL_II_FORCE_BUNDLE_BOOST=ON`). Anyway, you should verify that `detailed.log` shows that you are using an external `TBB`.

**Comment by LemonBoy68** [comment]
Add to the original:
-D TBB_LIBRARY=/opt/intel/oneapi/tbb/latest/lib/intel64/gcc4.8/libtbb.so \
-D TBB_DEBUG_LIBRARY=/opt/intel/oneapi/tbb/latest/lib/intel64/gcc4.8/libtbb_debug.so \
It seems that cmake can be made to find the external tbb? But the above error still occurs.
-----------------------
-- Include /media/ninghui/Aspect/packages/dealii/cmake/configure/configure_10_tbb.cmake
-- Found TBB_LIBRARY
-- Found TBB_DEBUG_LIBRARY
-- Found TBB_INCLUDE_DIR
-- Found TBB_INCLUDE_DIR
-- Processing TBB variables and targets
-- Processing TBB variables and targets - Done
-- The externally provided TBB library is older than version 4.2.0, which cannot be used with deal.II.
-- DEAL_II_WITH_TBB has unmet external dependencies.
-- Performing Test DEAL_II_HAVE_FLAG_Wno_parentheses
-- Performing Test DEAL_II_HAVE_FLAG_Wno_parentheses - Success
-----------------------

**Comment by bangerth** [comment]
We no longer bundle TBB, and we no longer require its presence either. Let's close this issue.
